contracts/src/tests/test_mcms.cairo

```diff
@@ -0,0 +1,1023 @@
+            group_parents.span(),
+            clear_root
+        );
+
+    match result {
+        Result::Ok(_) => panic!("expect 'group quorums/parents mismatch'"),
+        Result::Err(panic_data) => {
+            assert(*panic_data.at(0) == 'group quorums/parents mismatch', *panic_data.at(0));
+        }
+    }
+
+    // 5. test if group_quorum and group_parents not equal in length
+
+    // todo: replace with [0_u8; 32] in cairo 2.7.0
+    let mut group_quorums = ArrayTrait::new();
+    let mut i = 0;
+    while i < 32_usize {
+        group_quorums.append(0);
+        i += 1;
+    };
+
+    let result = mcms_safe
+        .set_config(
+            signer_addresses.span(),
+            signer_groups.span(),
+            group_quorums.span(),
+            group_parents.span(),
+            clear_root
+        );
+
+    match result {
+        Result::Ok(_) => panic!("expect 'group quorums/parents mismatch'"),
+        Result::Err(panic_data) => {
+            assert(*panic_data.at(0) == 'group quorums/parents mismatch', *panic_data.at(0));
+        }
+    }
+}
+
+#[test]
+#[feature("safe_dispatcher")]
+fn test_set_config_signers_group_out_of_bounds() {
+    // 6. test if one of signer_group #'s is out of bounds NUM_GROUPS
+    let (_, _, mcms_safe) = setup();
+
+    let signer_addresses = array![EthAddressZeroable::zero()];
+    let signer_groups = array![33];
+
+    let mut group_quorums = ArrayTrait::new();
+    let mut i = 0;
+    while i < 32_usize {
+        group_quorums.append(0);
+        i += 1;
+    };
+
+    let mut group_parents = ArrayTrait::new();
+    let mut i = 0;
+    while i < 32_usize {
+        group_parents.append(0);
+        i += 1;
+    };
+
+    let clear_root = false;
+
+    let result = mcms_safe
+        .set_config(
+            signer_addresses.span(),
+            signer_groups.span(),
+            group_quorums.span(),
+            group_parents.span(),
+            clear_root
+        );
+
+    match result {
+        Result::Ok(_) => panic!("expect 'out of bounds group'"),
+        Result::Err(panic_data) => {
+            assert(*panic_data.at(0) == 'out of bounds group', *panic_data.at(0));
+        }
+    }
+}
+
+#[test]
+#[feature("safe_dispatcher")]
+fn test_set_config_group_tree_malformed() {
+    // 7. test if group_parents[i] is greater than or equal to i (when not 0) there is revert
+    let (_, _, mcms_safe) = setup();
+
+    let signer_addresses = array![EthAddressZeroable::zero()];
+    let signer_groups = array![0];
+
+    let mut group_quorums = array![
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0
+    ];
+
+    let mut group_parents = array![
+        0,
```

Comment:
when we upgrade to cairo >= 2.7.0 we will be able to do something like [0_u8; 32].into() instead of having this ugly thing